contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+        uint256 _3crvBalanceFromDeposit = _3crvBalanceAfter - _3crvBalanceBefore;
+        uint256 _min3crvOut = (_daiBalance + _usdcBalance + _usdtBalance) * 1e18 / _3crvVirtualPrice * (BASIS_POINTS_GRANULARITY - depositMaxSlippageBasisPoints) / BASIS_POINTS_GRANULARITY;
+        require(_3crvBalanceFromDeposit >= _min3crvOut, "StableSwwapOperatorV1: 3pool deposit slippage too high");
+
+        // get the amount of tokens in the pool
+        (uint256 _3crvAmount, uint256 _feiAmount) = (
+            IStableSwap(pool).balances(_3crvIndex),
+            IStableSwap(pool).balances(_feiIndex)
+        );
+        // ... and the expected amount of 3crv in it after deposit
+        uint256 _3crvAmountAfter = _3crvAmount + _3crvBalanceAfter;
+
+        // get the usd value of 3crv in the pool
+        uint256 _3crvUsdValue = _3crvAmountAfter * _3crvVirtualPrice / 1e18;
+
+        // compute the number of FEI to deposit
+        uint256 _feiToDeposit = 0;
+        if (_3crvUsdValue > _feiAmount) {
+            _feiToDeposit = _3crvUsdValue - _feiAmount;
+        }
+
+        // deposit
+        if (_3crvBalanceAfter > 0 || _feiToDeposit > 0) {
+            // build parameters
+            uint256[2] memory _addLiquidityAmounts;
+            _addLiquidityAmounts[_feiIndex] = _feiToDeposit;
+            _addLiquidityAmounts[_3crvIndex] = _3crvBalanceAfter;
+
+            // approvals
+            IERC20(address(_fei)).approve(pool, _feiToDeposit);
+            // 3crv needs to reset allowance to 0, see CurveTokenV1.vy#L117 :
+            // assert _value == 0 or self.allowances[msg.sender][_spender] == 0
+            IERC20(_3crv).approve(pool, 0);
+            IERC20(_3crv).approve(pool, _3crvBalanceAfter);
+
+            // do deposit
+            uint256 _balanceBefore = IERC20(pool).balanceOf(address(this));
+            IStableSwap(pool).add_liquidity(_addLiquidityAmounts, 0);
+            uint256 _balanceAfter = IERC20(pool).balanceOf(address(this));
+
+            // emit event
+            uint256 _lpTotalSupply = IERC20(pool).totalSupply();
+            uint256 _3crvOut = (_3crvAmountAfter * (_balanceAfter - _balanceBefore)) / _lpTotalSupply;
+            uint256 _daiOut = IStableSwap(_3pool).calc_withdraw_one_coin(
+              _3crvOut, // LP tokens just deposited
+              0 // 3pool coin 0 = DAI
+            );
+            emit Deposit(msg.sender, _daiOut);
+        }
+
+        // burn remaining FEI
+        uint256 _feiBalance = _fei.balanceOf(address(this));
+        _fei.burn(_feiBalance);
+    }
+
+    /// @notice withdraw DAI from the deposit. This will remove liquidity in the
+    /// current pool's proportion, burn the FEI part of it, and then use the 3crv
+    /// to withdraw liquidity of the 3pool in DAI.
+    /// note: because of slippage & fees on withdraw, the amount of DAI out is
+    ///        not exactly the expected amountUnderlying provided as input.
+    /// @param amountUnderlying of tokens withdrawn (0 = all)
```

Comment:
I don't like the 0=all logic. If anything it should be withdraw min(balance, amountUnderlying). Either way we have a ratioPCVController that can do the total withdraw so I'd rather remove this logic

Reply:
I'll remove the withdraw 0 logic